pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -302,30 +304,46 @@ func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// nsx bug, if set LoadBalancerVpcEndpoint.Enabled to false, when read this VPC back,
 	// LoadBalancerVpcEndpoint.Enabled will become a nil pointer.
 	if lbProvider == vpc.AVILB && createdVpc.LoadBalancerVpcEndpoint != nil && createdVpc.LoadBalancerVpcEndpoint.Enabled != nil && *createdVpc.LoadBalancerVpcEndpoint.Enabled {
-		path, cidr, err = r.Service.GetAVISubnetInfo(*createdVpc)
+		aviSubnetPath, aviSECIDR, err = r.Service.GetAVISubnetInfo(*createdVpc)
 		if err != nil {
-			log.Error(err, "Failed to read LB Subnet path and CIDR", "VPC", createdVpc.Id)
+			log.Error(err, "Failed to read AVI LB Subnet path and CIDR", "VPC", createdVpc.Id)
 			state := &v1alpha1.VPCState{
 				Name:                    *createdVpc.DisplayName,
 				DefaultSNATIP:           snatIP,
 				LoadBalancerIPAddresses: "",
 				PrivateIPs:              privateIPs,
 			}
-			r.StatusUpdater.UpdateFail(ctx, networkInfoCR, err, fmt.Sprintf("Failed to read LB Subnet path and CIDR, VPC: %s", *createdVpc.Id), setNetworkInfoVPCStatusWithError, state)
+			r.StatusUpdater.UpdateFail(ctx, networkInfoCR, err, fmt.Sprintf("Failed to read AVI LB Subnet path and CIDR, VPC: %s", *createdVpc.Id), setNetworkInfoVPCStatusWithError, state)
 			setNSNetworkReadyCondition(ctx, r.Client, req.Namespace, nsMsgVPCAviSubnetError.getNSNetworkCondition(err))
 			return common.ResultRequeueAfter10sec, err
 		}
+		lbIP = aviSECIDR
+	} else if lbProvider == vpc.NSXLB {
+		nsxLBSNATIP, err = r.Service.GetNSXLBSNATIP(*createdVpc)
```

Comment:
could we return the VPC Tier1 LB SNAT IP in CreateOrUpdateVPC func when checking the realization state of VPC? 
we can avoid a new NSX API call to get VPC realization state again.

Reply:
could you also help check if VPC is created with AVI Subnet, we still can get the LB SNAT IP right?

Reply:
I checked the logic of checking VPC realization state in CreateOrUpdateVPC. The logic for checking VPC realization state  is different with getting VPC LB SNAT IP. I feel that we had better not mix the two logic in one piece. Indeed, we can reduce one time API call but it's not good for code clean style and code maintenance.

Reply:
>could you also help check if VPC is created with AVI Subnet, we still can get the LB SNAT IP right?
Yes, I checked one VPC AVI testbed, we can get AVI SE subnet CIDR:loadBalancerIPAddresses: 172.17.0.0/26
 ```
kubectl get networkinfo ns-1 -nns-1 -o yaml
apiVersion: crd.nsx.vmware.com/v1alpha1
kind: NetworkInfo
metadata:
  creationTimestamp: "2025-02-21T05:21:43Z"
  generation: 2
  name: ns-1
  namespace: ns-1
  resourceVersion: "2150989"
  uid: 94f297ee-c230-4f45-814d-02807d25a977
vpcs:
- defaultSNATIP: 192.168.0.8
  loadBalancerIPAddresses: 172.17.0.0/26
  name: ns-1_94f297ee-c230-4f45-814d-02807d25a977
  privateIPs:
  - 172.17.0.0/16
```